src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -55,19 +55,36 @@
                                TargetMonikers="@(TestNugetTargetMoniker)">
       <Output TaskParameter="ResolvedCopyLocalItems" ItemName="TestCopyLocal" />
     </PrereleaseResolveNuGetPackageAssets>
-
+    
     <ItemGroup Condition="'$(TestWithCore)' != 'true'">
 
       <_TestCopyLocalXunitFiles Include="$(PackagesDir)xunit.runner.console\2.1.0\tools\*.exe" />
       <_TestCopyLocalXunitFiles Include="$(PackagesDir)xunit.runner.console\2.1.0\tools\*.dll" />
-      <_TestCopyLocalXunitFiles Condition="'$(TestAppX)' == 'true'" Include="$(PackagesDir)microsoft.xunit.runner.uwp\**\*.*" />
 
       <TestCopyLocal Include="@(_TestCopyLocalXunitFiles)" >
         <NuGetPackageId>%(FileName)</NuGetPackageId>
         <NuGetPackageVersion>2.1.0</NuGetPackageVersion>
         <Private>false</Private>
       </TestCopyLocal>
     </ItemGroup>
+      
+    <!-- Select a UWP Runner Version from the package directory-->
+    <CreateItem Condition="'$(TestAppX)' == 'true'" Include="$(PackagesDir)microsoft.xunit.runner.uwp\*\*.*">
+    <Output TaskParameter="Include" ItemName="_UwpRunnerFiles" />
+    </CreateItem>
+    <RemoveDuplicates Condition="'$(TestAppX)' == 'true'" Inputs="@(_UwpRunnerFiles->'%(RelativeDir)')">
+    <Output TaskParameter="Filtered" ItemName="_ItemGroupUwpRunnerVersionCandidates"/>
+    </RemoveDuplicates>
+    <ItemGroup Condition="'$(TestAppX)' == 'true'">
+      <_UwpRunnerDirectory Include="%(_ItemGroupUwpRunnerVersionCandidates.Identity)" Condition="'@(_UwpRunnerVersion->Count())'!=1" />
+    </ItemGroup>
+    
+    <ItemGroup Condition="'$(TestWithCore)' != 'true'">
+      <_TestCopyLocalXunitFiles Condition="'$(TestAppX)' == 'true'" Include="$(_UwpRunnerDirectory)\*.*" />
```

Comment:
$(_UwpRunnerDirectory) is assuming you only found one item in your item group above.